### PR TITLE
Remove `DepthWrappingVertex<T>`

### PR DESCRIPTION
**Pull request by smoogipoo**
This is a breaking change so I suggest applying it as late as possible as there will be other breaking changes.

Taking the opportunity to make a bit of a breaking change while cleaning this up. The existence of this object is a bit weird - it's as if it's an implementation detail yet is not treated as an implementation detail - you need to be very well aware that this is a thing when implementing renderers.

---

### `TexturedVertex2D` must be initialised with an `IRenderer`.

The default constructor for this type has been obsoleted and will now generate a compiler error.

### CODE REVIEW

**Review by bdach — COMMENTED** [review]
Code looks quaint enough. Will run some platform testing later but I don't expect surprises.

File: osu.Framework/Graphics/Rendering/Vertices/TexturedVertex2D.cs [hunk 1]
```diff
@@ -27,11 +27,27 @@ public struct TexturedVertex2D : IEquatable<TexturedVertex2D>, IVertex
...
+
+        [Obsolete("Initialise this type with an IRenderer instead", true)]
+        public TexturedVertex2D()
+        {
+            this = default;
+        }
+
+        public TexturedVertex2D(IRenderer renderer)
+        {
+            this = default;
```
[the remark this answers is not in the document]
Comment (peppy) [reply osu.Framework/Graphics/Rendering/Vertices/TexturedVertex2D.cs:41]: Might be legal, but I'd have no idea what this means nor how to quickly find the documentation for what it means 😅 . Might be worth a comment?

[the remark this answers is not in the document]
Comment (bdach) [reply osu.Framework/Graphics/Rendering/Vertices/TexturedVertex2D.cs:41]: It's just a shorthand to initialise all members of the struct to default values for their type, and initialisation is required for structs. Easier to write `this = default;` than to do this per member in both ctors.

[the remark this answers is not in the document]
Comment (bdach) [reply osu.Framework/Graphics/Rendering/Vertices/TexturedVertex2D.cs]: Inline comment added in https://github.com/ppy/osu-framework/pull/5943/commits/3da418703b2cec4853480da36376185787303535

**Review by peppy — APPROVED** [review]

**Review by bdach — APPROVED** [review]

**Pull request merged**